docs/components/configurationPanel.md

```diff
@@ -0,0 +1,224 @@
+| `message` | `{Object}` | Contains the list of actions supported by the selected gamepad in two separate child objects, namely `buttons` and `axes` input. The key name should match with the name of the navigation invoker in the [`inputMapper`](inputMapper.md) component. These actions are used in the **Action** dropdown of the configuration panel. |
+| `actions` | `{Object}` | Specifies whether the selected action supports `speedFactor`, `invert`, or `background` configuration options. |
+
+These options can be provided to the component in the following ways:
+
+``` javascript
+// Either pass options inside the object as an argument while creating an instance of configurationPanel component.
+var configurationPanelInstanceOne = gamepad.configurationPanel({
+    description: {
+        buttons: { "16": "Button 16: Badge Icon Button" },
+        axes: { "0": "Left Thumbstick Horizontal Direction" }
+    },
+    message: {
+        buttons: {
+            null: "None",
+            scrollLeft: "Scroll left",
+            scrollRight: "Scroll right"
+        },
+        axes: {
+            null: "None",
+            thumbstickTabbing: "Focus on the previous/next element"
+        }
+    },
+    actions: {
+        speedFactorOption: ["scrollLeft", "scrollRight", "thumbstickTabbing"],
+        invertOption: ["thumbstickTabbing"]
+    }
+});
+
+// Otherwise pass it as default options in a custom component using the configurationPanel grade.
+fluid.defaults("my.configurationPanel.grade", {
+    gradeNames: ["gamepad.configurationPanel"],
+    cutoffValue: 0.10,
+    scrollInputMultiplier: 25
+});
+var configurationPanelInstanceTwo = my.configurationPanel.grade({
+    description: {
+        buttons: { "16": "Button 16: Badge Icon Button" },
+        axes: { "0": "Left Thumbstick Horizontal Direction" }
+    },
+    message: {
+        buttons: {
+            null: "None",
+            scrollLeft: "Scroll left",
+            scrollRight: "Scroll right"
+        },
+        axes: {
+            null: "None",
+            thumbstickTabbing: "Focus on the previous/next element"
+        }
+    },
+    actions: {
+        speedFactorOption: ["scrollLeft", "scrollRight", "thumbstickTabbing"],
+        invertOption: ["thumbstickTabbing"]
+    }
+});
+```
+
+## Invokers
+
+### `{configurationPanel}.createMenu()`
+
+- Returns: Nothing.
+
+Creates a separate configuration menu for each gamepad input, as mentioned in the `description` option and is triggered
+when the component is created. This invoker calls other invokers to create the configuration options and to listen to
+various interaction-based events and changes on the configuration panel.
+
+### `{configurationPanel}.createInputActionDropdown(inputIdentifier, configMenu, inputType, currentValue)`
+
+- `inputIdentifier {String}` A description of the gamepad input type and its index. For example, "button-0".
```

Comment:
Sorry for not catching this earlier,  but it seems less than ideal to in essence encode two pieces of data in a string like this using your own convention.  I'd suggest filing a ticket to cover converting this to use the same dot-notation supported by `fluid.get` and `fluid.put`, i.e. `button.0` instead of `button-0`.

You would replace whatever is parsing the string into the two pieces of data (or combining it) with `fluid.get` or `fluid.set` rather than string splitting and concatenation.

Reply:
Thanks for pointing this out! I've filed the ticket [here](https://github.com/fluid-lab/gamepad-navigator/issues/58).